metric/emitter/prometheus.go

```diff
@@ -0,0 +1,193 @@
+package emitter
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/concourse/atc/db"
+	"github.com/concourse/atc/metric"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type PrometheusEmitter struct {
+	buildsStarted     prometheus.Counter
+	buildsFinished    prometheus.Counter
+	buildsSucceeded   prometheus.Counter
+	buildsErrored     prometheus.Counter
+	buildsFailed      prometheus.Counter
+	buildsAborted     prometheus.Counter
+	buildsFinishedVec *prometheus.CounterVec
+	buildDurationsVec *prometheus.HistogramVec
+}
+
+type PrometheusConfig struct {
+	BindAddress string `long:"prometheus-bind-address" description:"Address to listen on to expose Prometheus metrics."`
```

Comment:
Could this be renamed to `BindIP` and `prometheus-bind-ip` for consistency with the rest of the ATC? Thanks.